### ISSUE TITLE
Disable creation of texturepacks, mods, etc. folders based on version and loader

### ISSUE THREAD

**Issue by vrad-exe** [PrismLauncher/PrismLauncher#620]
### Role

I use Prism Launcher to manage multiple modded and vanilla Minecraft instances

### Suggestion

When the instance properties window is open, folders will automatically be created under `<instance>/minecraft/` for texture packs, saves, mods, etc. This should be changed to check the selected version/loader, and only create the folders that are actually used.

### Benefit

So instance folders aren't cluttered up with irrelevant folders, like texture packs in post 1.6 versions

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

The folders should only be created under these conditions:
- `coremods`: Forge installed, game version <= 1.5.2
- `mods`: Any mod loader installed
- `resourcepacks`: Game version >= 13w24a
- `saves`: Game version >= [Infdev 20100327](https://minecraft.fandom.com/wiki/Java_Edition_Infdev_20100327) (before this, saving used a file picker instead of a specific folder)
- `screenshots`: Game version >= Alpha 1.2.0
- `texturepacks`: Game version < 13w24a

`shaderpacks` is a weird case, since this folder is entirely dependent on specific mods being installed in order to do anything, and I can't think of a way to know if those mods are present besides using a hardcoded list of known shader mods. Perhaps just never create it automatically and wait for the game to do so if a shader mod is present.

Note that this would never delete the folders if they already exist, only determine whether or not they should be automatically created if they don't. The instance properties tabs should also be conditionally visible based on this same system, but if the corresponding folder already exists and has things in it, the tab would be forced to appear anyway.

Of course, this would require telling the launcher which versions/loaders actually support these features, ideally in a non-hardcoded way. The launcher already seems to be aware of this for resource/texture packs though (swapping which tab is displayed), so I can't imagine it being too complex to expand that to other folders/tabs. At the very least, this should apply to resource/texture packs, and coremods should be disabled since that's only used by very old versions of Forge.

Hopefully I explained this well.

**Comment by DioEgizio** [comment]
> shaderpacks is a weird case, since this folder is entirely dependent on specific mods being installed in order to do anything, and I can't think of a way to know if those mods are present besides using a hardcoded list of known shader mods. Perhaps just never create it automatically and wait for the game to do so if a shader mod is present.

i can see an issue with that approach: you can't use the shaderpack tab until you loaded the game then

**Comment by vrad-exe** [comment]
Yeah, the alternative I thought of was just adding the shader packs tab/folder if any mod loader or jar mod is installed (so same as the mods folder pretty much). Could also try and detect shader mods, but that seems unreliable.

**Comment by DioEgizio** [comment]
Shadermod detection has 2 issues:
- a new shadermod can come out at any time
- optifine didn't have mod metadata info attached to it before 1.13

**Comment by vrad-exe** [comment]
Yeah that's why it would be unreliable. Having it be the same condition as mods is probably the best option.

**Comment by TheKodeToad** [comment]
OptiFine isn't always installed as a mod.

**Comment by DioEgizio** [comment]
yeah also that :/

**Comment by Continous** [comment]
Could just not generate the shaderpack folder until the user attempted to actually interact with it in some manner. IE, if the user attempts to open the folder or add a shaderpack, the folder will be created, but not until then.

**Comment by rorybd** [comment]
It would be much nicer if Prism only created these folders when and as they are actually needed
